### ISSUE TITLE
Conan >=1.3 breaks multi-package support

### ISSUE THREAD

**Issue by bschindler** [conan-io/conan#2981]
We use the N configs -> 1 package approach (still, we didn't manage to switch to a 1 config 1 package approach due to technical difficulties). So a typical package_info looks as follows:

```
def package_info(self):
        if self.settings.os != "Linux":
            self.cpp_info.debug.libs = ['benchmarkd']
            self.cpp_info.release.libs = ['benchmark']
        else:
            self.cpp_info.libs =  ['benchmark']
```
However, conan is not able to locate the libraries at all, I get the output: 

Library benchmarkd not found in package, might be system one
Library benchmark not found in package, might be system one

I looked at the generated conanbuildinfo.cmake, and noted that starting with conan 1.3, the following call is made: 

```
conan_package_library_targets("${CONAN_LIBS_GBENCHMARK_DEBUG}" "${CONAN_LIB_DIRS_GBENCHMARK_DEBUG}"
                                  CONAN_PACKAGE_TARGETS_GBENCHMARK_DEBUG "" "debug" GBenchmark)
```
But the variable CONAN_LIB_DIRS_GBENCHMARK_DEBUG is not set (but used to be set on conan 1.2). 

I have the following: 

set(CONAN_LIB_DIRS_GBENCHMARK "C:/Users/benjamin.schindler/.conan/data/GBenchmark/1.4.0/virtamed/testing/package/98aaa1d6a87f0b802f745a8ec2e8347f131e031c/lib")
set(CONAN_LIB_DIRS_GBENCHMARK_DEBUG )
set(CONAN_LIB_DIRS_GBENCHMARK_RELEASE )

With using 1.2.3, those variables are set: 

set(CONAN_LIB_DIRS_GBENCHMARK "C:/Users/benjamin.schindler/.conan/data/GBenchmark/1.4.0/virtamed/testing/package/98aaa1d6a87f0b802f745a8ec2e8347f131e031c/lib")
set(CONAN_LIB_DIRS_GBENCHMARK_DEBUG "C:/Users/benjamin.schindler/.conan/data/GBenchmark/1.4.0/virtamed/testing/package/98aaa1d6a87f0b802f745a8ec2e8347f131e031c/lib")
set(CONAN_LIB_DIRS_GBENCHMARK_RELEASE "C:/Users/benjamin.schindler/.conan/data/GBenchmark/1.4.0/virtamed/testing/package/98aaa1d6a87f0b802f745a8ec2e8347f131e031c/lib")

This issue currently blocks us from going beyond 1.2.3 (looks like a regression issue to me)

**Comment by memsharded** [comment]
Yes, could be a regression, checking it right now, thanks for reporting

**Comment by memsharded** [comment]
Could you post your full ``package_info()`` method. There was a bug that was corrected related to this, it would be useful to check if you are also settings other paths. Thanks!

**Comment by memsharded** [comment]
The related PR is: https://github.com/conan-io/conan/pull/2740

**Comment by bschindler** [comment]
That's the full package_info(), pretty boring tbh. 
```
def package_info(self):
        if self.settings.os != "Linux":
            self.cpp_info.debug.libs = ['benchmarkd',  'output_test_helperd']
            self.cpp_info.release.libs = ['benchmark', 'output_test_helper']
        else:
            self.cpp_info.libs =  ['benchmark', 'output_test_helper']

        if self.settings.os != "Linux":
            self.cpp_info.libs.append("shlwapi")
```

**Comment by memsharded** [comment]
This should be fixed and released in 1.4.2, please check and report otherwise. Many thanks!
